mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/programmatic/PropertyMappingVectorFieldStep.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.definition.programmatic;
+
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+
+/**
+ * TODO: vector : docs
+ */
+public interface PropertyMappingVectorFieldStep extends PropertyMappingStep {
+
+	PropertyMappingVectorFieldStep projectable(Projectable projectable);
+
+	PropertyMappingVectorFieldStep vectorSimilarity(VectorSimilarity vectorSimilarity);
+
+	PropertyMappingVectorFieldStep dimension(Integer dimension);
```

Comment:
If the dimension is mandatory (and I think it is?) you should probably pass it to `propertyMappingStep#vectorField` instead?